INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift

```diff
@@ -13,67 +13,151 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
     
     var validateEmail = Validate()
     
-    @IBOutlet weak var logoImageView: UIImageView!
+    let spacingBetweenCells: CGFloat = 40
+    var heightTableView: CGFloat = 540
     
-    @IBOutlet weak var blueLineView: UIView!
+    @IBOutlet weak var forgotTableView: UITableView!
     
-    @IBOutlet weak var recoveryLabel: UILabel!
-    @IBOutlet weak var explanationTextLabel: UILabel!
+    @IBOutlet weak var tableViewTopConstraint: NSLayoutConstraint!
+    @IBOutlet weak var tableViewBottomConstraint: NSLayoutConstraint!
     
-    @IBOutlet weak var emailTextField: UITextField!
-    
-    @IBOutlet weak var invalidTextLabel: UILabel!
-    @IBOutlet weak var sendButton: UIButton!
-
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.recoveryLabel.text = "passRecovery".localized
-        self.explanationTextLabel.text = "textRecovery".localized
+        forgotTableView.delegate = self
+        forgotTableView.dataSource = self
+        
+        registerCells()
+        
+        navigationController?.navigationBar.barTintColor = UIColor.white
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(true)
         
-        setupInvalidTextLabel()
-        setupEmailTextField()
-        setupSendButton()
+        startMonitoringKeyboard()
         
     }
     
-    func setupInvalidTextLabel() {
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(true)
         
-        let title = ""
-        invalidTextLabel.text = title
-        invalidTextLabel.textColor = .red
+        stopMonitoringKeyboard()
 
     }
     
-    func setupEmailTextField() {
+    override func animateKeyboardAppearance(height: CGFloat) {
+        
+        let heightView = UIScreen.main.bounds.height
+        let differentHeight = heightView - heightTableView
+        
+        var resultHeightConstreint: CGFloat
+        
+        if (heightTableView - differentHeight) > height {
+            resultHeightConstreint = -height - (spacingBetweenCells / 2)
+        } else {
+            resultHeightConstreint = differentHeight - heightTableView
+        }
 
-        emailTextField.layer.cornerRadius = 4
-        emailTextField.borderStyle = .line
-        emailTextField.keyboardAppearance = .dark
-        emailTextField.backgroundColor = .white
-        emailTextField.keyboardType = .emailAddress
-        emailTextField.placeholder = "Enter your Email"
-        emailTextField.clearButtonMode = .whileEditing
+        forgotTableView.beginUpdates()
+        tableViewTopConstraint.constant = resultHeightConstreint
+        view.layoutSubviews()
+        forgotTableView.endUpdates()
```

Comment:
scroll tableView to submit button using forgotTableView.scrollToRow(at: IndexPath(row: submitButtonRow, section: 0), at: .bottom, animated: true)